data-processing/entities/definitions/model/model/modeling_joint_roberta.py

```diff
@@ -49,23 +53,26 @@ def __init__(self, config: RobertaConfig, args: Any, intent_label_lst: List[str]
             hidden_size += self.num_acronym_labels
 
         self.custom_pooler = Pooler(hidden_size=hidden_size)
-        self.intent_classifier = IntentClassifier(
-            hidden_size, self.num_intent_labels, args.dropout_rate
-        )
-        self.slot_classifier = SlotClassifier(
-            hidden_size, self.num_slot_labels, args.dropout_rate
-        )
+        for pred_type in tasks:
+            self.intent_classifiers[pred_type] = IntentClassifier(hidden_size, self.num_intent_labels_dict[pred_type], args.dropout_rate)
 
-        if args.use_crf:
-            self.crf = CRF(num_tags=self.num_slot_labels, batch_first=True)
+        for pred_type in tasks:
+            self.slot_classifiers[pred_type] = SlotClassifier(hidden_size, self.num_slot_labels_dict[pred_type], args.dropout_rate)
+
+            if args.use_crf:
+                self.crfs[pred_type] = CRF(num_tags=self.num_slot_labels_dict[pred_type], batch_first=True)
+        
+        self.intent_classifiers = nn.ModuleDict(self.intent_classifiers)
```

Comment:
out of curiosity, what does wrapping these attributes in moduledicts do?

Reply:
This is the pytorch API for having dynamic layers that are not explicitly defined. As you can see here, the layers are initialized based on the tasks.